src/main/java/edu/wpi/grip/core/SocketHints.java

```diff
@@ -0,0 +1,105 @@
+package edu.wpi.grip.core;
+
+
+import org.bytedeco.javacpp.opencv_core.Size;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Point;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+
+/**
+ * Create default socket hints
+ */
+public final class SocketHints {
+
+    private SocketHints() { /* no op */ }
+
+    public static final class Outputs {
+        private Outputs() { /* no op */ }
+
+        public static SocketHint<Mat> createMatSocketHint(final String identifier) {
+            return SocketHints.createMatSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Point> createPointSocketHint(final String identifier) {
+            return SocketHints.createPointSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Size> createSizeSocketHint(final String identifier) {
+            return SocketHints.createSizeSocketHint(identifier, true);
+        }
+
+        public static SocketHint<Boolean> createBooleanSocketHint(final String identifier, boolean defaultValue) {
+            return new SocketHint.Builder(Boolean.class).identifier(identifier).initialValue(defaultValue).publishable(true).build();
+        }
+
+        public static SocketHint<Number> createNumberSocketHint(final String identifier, Number defaultValue) {
+            return new SocketHint.Builder(Number.class).identifier(identifier).initialValue(defaultValue).publishable(true).build();
+        }
+
+    }
+
+    public static SocketHint<Mat> createMatSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Mat.class, Mat::new, withDefault).build();
+    }
+
+    public static SocketHint<Size> createSizeSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Size.class, Size::new, withDefault).build();
+    }
+
+
+    public static SocketHint<Point> createPointSocketHint(final String identifier, final boolean withDefault) {
+        return createObjectSocketHintBuilder(identifier, Point.class, Point::new, withDefault).build();
+    }
+
+    public static SocketHint<Number> createNumberSliderSocketHint(final String identifier, final Number number,
```

Comment:
Since the domain for sliders has to be two values, it would be safer to make this take two numbers instead of an array